code/script.js

```diff
@@ -0,0 +1,136 @@
+const MY_REPOS = 'https://api.github.com/users/ruruahn/repos'
+const projectsContainer = document.getElementById("projects")
+const commitsContainer = document.getElementById("commits")
+const user = 'ruruahn'
+const USER_URL = `https://api.github.com/users/${user}`
+const userContainer = document.getElementById("userProfile")
+
+const checkHTML = document.getElementById("HTML")
+const checkCSS = document.getElementById("CSS")
+const checkJavaScript = document.getElementById("JavaScript")
+let selectedLanguages = ['HTML', 'CSS', 'JavaScript']
+let chartDrawn = false
+
+//Create link images to social media
+const createSocialMediaImg = (url, alt) => {
+    let image = document.createElement("img")
+    image.src = url
+    image.alt = alt
+    return image
+}
+
+const githubIcon = createSocialMediaImg("./img/github.png", "github")
+
+//Function to fetch and display profile information
+const userProfile = () => {
+    fetch(USER_URL)
+        .then(res => res.json())
+        .then(data => {
+            console.log(data)
+            userContainer.innerHTML = `
+        <img src="${data.avatar_url}" class="avatar"/>
+        <p class="full-name">${data.name}</p>
+        <p class="username">Username: ${data.login}</p>
+        <p class="location">${data.location}</p>
+        <a href="${data.html_url}" target="blank"><img src="${githubIcon.src}" alt="${githubIcon.alt}"  class="github-icon"/></a>
+        `
+        })
+}
+
+//Function to fetch and display my repos
+const fetchRepos = () => {
+    fetch(MY_REPOS)
+        .then((res) => res.json())
+        .then((data) => {
+            projectsContainer.innerHTML = `` //Clear innerHTML to prevent duplicated repos
+            const forkedRepos =
+                data.filter(repo => {
+                    return repo.fork && repo.name.startsWith("project-") && selectedLanguages.includes(repo.language)
+                })
+            //Function to sort repos by latest date
+            forkedRepos.sort(function (a, b) {
+                return new Date(b.pushed_at) - new Date(a.pushed_at)
+            })
+            forkedRepos.forEach(repo => {
+                projectsContainer.innerHTML += `
+                      <div class="repo" id=${repo.name}>
+                        <p>Name: ${repo.name} </p> 
+                        <a href="${repo.html_url}" target="blank">Go to repo!</a> 
+                        <p>Branch: ${repo.default_branch}</p>
+                        <p>Latest push: ${new Date(repo.pushed_at).toDateString()}</p>
+                        <p id="commit-${repo.name}">Amount of commits: </p> 
+                      </div>
+                        `
+            })
+
+            fetchPullRequests(forkedRepos)
+
+            //Only draw the chart once
+            if (!chartDrawn) {
```

Comment:
This is very smart approach ! Great that it is drawn only once per session . Great job!